### ISSUE TITLE
Elpis 1.4.7 cannot set the MediaPlayPause key as a hotkey

### ISSUE THREAD

**Issue by Wires77** [adammhaile/Elpis#40]
It comes pre-set, but you can't set it as a new hotkey it would appear.  It could just be my keyboard, but I'm not sure.

**Comment by seliver** [comment]
What's a GlobalMediaKey exactly?

**Comment by DocBennett** [comment]
Similar/Linked as #41 

Wild guess/search: [Installing a global hot key with C#](http://www.liensberger.it/web/blog/?p=207)

**Comment by seliver** [comment]
Can you please send a printscreen of your Elpis settings screen?

**Comment by Wires77** [comment]
This is simply a fresh install of 1.4.7 in the hotkey section.  I updated the title to reflect the actual hotkey name.

![elpis 1 4 7 settings](https://cloud.githubusercontent.com/assets/1209372/6489080/db0c2b32-c258-11e4-83a6-1314f74b1ed6.png)

**Comment by seliver** [comment]
Try deleting the other one using MediaPlayPause. Until that, you can't use it again on another hotkey.

**Comment by Wires77** [comment]
Those are just the defaults.  Even if I delete that Play/Pause hotkey I still can't set another with MediaPlayPause

**Comment by seliver** [comment]
That's strange. Here on my setup I can do that with no problem. Maybe 
because of your keyboard? What's your model? Do those buttons work with 
windows media player or vlc?

Alexey

**Comment by Wires77** [comment]
Hmm...yeah the button works with WMP and it worked on 1.4.5

The model of keyboard is a Microsoft Natural Ergonomic 4000.  It may have to do with IType, the program that makes some of the extra keyboard keys work.  Messing around with it has now rendered 1.4.5 to have the same problem, so I'll have to give this another go on my home computer to see if it has the problem there.

**Comment by DocBennett** [comment]
Wiress77 - does it work with a modifyer (Ctrl/Alt/Shift) in combo with the media key press?

**Comment by seliver** [comment]
The strange thing is that i have a Microsoft keyboard that has got a play/pause button, and it is detected just by itself.

**Comment by Wires77** [comment]
Okay, so I figured out that if I stop the itype.exe process, everything seems to magically work again...I could've swore I tested that earlier today, so I'm going to fiddle with it some more.  For my purposes, this seems to work now though.

**Comment by seliver** [comment]
Great to hear that! 
So now @DocBennett, how's Elpis behaving there?

**Comment by DocBennett** [comment]
I am using a Logitech G15 keyboard. I am not currently using its profiler app "Logitech Gaming Software" at all, not running. 

I tested the built-in media keys with VLC and would't you know it - they too do not detect the keypress unless I use a modifyer key. Ctrl/Alt/Shift + MediaPlayPause picks up fine but not Media Play Pause alone. 

I would have to imagine that this is just unique to theis keyborad when its provided profile software isn't being used. I have a ticket open with Logitech about their profiling software not working properly for me (prevents Windows 7 from detecting Logitech G930 headset properly).

If other testing/users are having no further issudes with hotkeys detecting media keys then I wll stand down.

**Comment by seliver** [comment]
Great, then it looks like Elpis is working fine now. Will wait a few 
more days to open the auto-update.

Alexey

**Comment by keathmilligan** [comment]
I'm not sure I understand the resolution of this. Global keys are definitely not working correctly.

Install 1.4.5: global media keys work as expected
Install 1.4.7: global media keys do not work, nor can the play/pause key be reassigned if it is deleted.

This is with a basic Microsoft wireless 800 keyboard/mouse combo running the "Mouse and Keyboard Center" software - everything at default settings.

**Comment by seliver** [comment]
Try assigning the keys without the "mouse and keyboard center" running. 
If in doubt, try and see if the assigning works on vlc.

On 11/03/2015 11:40, keathmilligan wrote:
>
> I'm not sure I understand the resolution of this. Global keys are 
> definitely not working correctly.
>
>
>
> Install 1.4.5: global media keys work as expected
>
> Install 1.4.7: global media keys do not work, nor can the play/pause 
> key be reassigned if it is deleted.
>
>
>
> This is with a basic Microsoft wireless 800 keyboard/mouse combo 
> running the "Mouse and Keyboard Center" software - everything at 
> default settings.
>
>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adammhaile/Elpis/issues/40#issuecomment-78247782>.
>
>
>
>
>
>
>
>
>
>
>
>


-- 
Alexey

**Comment by keathmilligan** [comment]
OK, I see what is happening now.

Before messing with assignments or shutting anything down, the Play/Pause key works globally with Elpis 1.4.5, iTunes and Windows Media Player. It works locally with VLC, but not globally. Elpis 1.4.7 will not work. I notice that you also cannot assign the Play/Pause key in VLC.

Shutting down itype.exe fixes the assignment issue in both VLC and Elpis and gets things working again. I guess this is an OK work-around since I don't need itype on this machine, but Elpis should have a fallback mode to respond to media keys the way 1.4.5 did.

**Comment by seliver** [comment]
OK, it's a thing to consider if many users require such feature.
But thanks for reporting Keath :)

On 11/03/2015 15:00, keathmilligan wrote:
>
> OK, I see what is happening now.
>
>
>
> Before messing with assignments or shutting anything down, the 
> Play/Pause key works globally with Elpis 1.4.5, iTunes and Windows 
> Media Player. It works locally with VLC, but not globally. Elpis 1.4.7 
> will not work. I notice that you also cannot assign the Play/Pause key 
> in VLC.
>
>
>
> Shutting down itype.exe fixes the assignment issue in both VLC and 
> Elpis and gets things working again. I guess this is an OK work-around 
> since I don't need itype on this machine, but Elpis should have a 
> fallback mode to respond to media keys the way 1.4.5 did.
>
>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/adammhaile/Elpis/issues/40#issuecomment-78280103>.
>
>
>
>
>
>
>
>
>
>
>
>


-- 
Alexey

**Comment by seliver** [comment]
Added a warning message on the changelog about this.

**Comment by maltschul** [comment]
I have the same problem that if itype.exe is running, Elpis won't accept the global media play/pause command. If I kill itype, it works. Start itype again, no worky.

**Comment by seliver** [comment]
But are you saying to assign those keys or use them, while the itype is 
active?

Alexey

**Comment by maltschul** [comment]
I'm saying that any time itype.exe is running, Elpis doesn't not respond to the Play/Pause (or Next Track) media keys. This includes trying to assign the key is settings or while music is playing.

**Comment by seliver** [comment]
Have you tried doing the same on vlc, p.ex.?

Alexey

**Comment by keathmilligan** [comment]
This will be a problem for users of Microsoft mice and keyboards moving forward. Previously, Microsoft distributed IntelliType and IntelliPoint separately and you could uninstall IntelliType if you didn't need it. But now they've dropped support for IntelliType/IntelliPoint in favor of "Mouse and Keyboard Center". With it, you can't selectively remove the itype component and of course if you remove the whole thing, you'll lose mouse functionality like button assignments and such.

**Comment by seliver** [comment]
But as i see, this problem is not affecting only elpis, but also other opensource projects. So it's probable that itype consumes all media key inputs and doesn't let other programs use it.

**Comment by keathmilligan** [comment]
Agreed. itype shouldn't work the way it does, but this is Microsoft :)

**Comment by grojguy** [comment]
FYI, for anyone using Logitech keyboard/mouse.  I was having the same problem, and I noticed that I had the Logitech SetPoint Control Center running.  As soon as I closed it, global hotkey functionality was working again.

**Comment by seliver** [comment]
Em 19/05/2015 21:06, "grojguy" <notifications@github.com> escreveu:
>
> FYI, for anyone using Logitech keyboard/mouse. I was having the same
problem, and I noticed that I had the Logitech SetPoint Control Center
running. As soon as I closed it, global hotkey functionality was working
again.

**Comment by grojguy** [comment]
Actually... upon further testing, I observe that Elpis is unique in **not** recognizing media control keys when Logitech SetPoint Control Center is running (which, according to earlier comments here, is also the case with the Microsoft "Mouse and Keyboard Center") .  

To clarify, I tested several other media player apps (Spotify, Windows Media Player, MPC HC), all of which respond correctly to media control keys with or without Logitech SetPoint Control Center running. (NOTE: It appears to be a common requirement that the "vendor software" e.g. Logitech SetPoint be running when opening the media player application.)

So it is looking more like this may indeed be an issue unique to Elpis.

**Comment by seliver** [comment]
Have you tried on VLC?

The thing about WMP and Spotify is that there you cannot assign hotkeys 
at your will, you just get what's default.
I don't know about MPC HC though.

Anyway at the moment I'm not seeing a solution for this.

Alexey

Às 23:55 de 15/06/2015, grojguy escreveu:

**Comment by keathmilligan** [comment]
Well, it worked in 1.4.5, so something about the way it handled the media keys in different than more recent versions. One, admittedly more complicated, possible way to deal with this would be to check for the actions being assigned to the common media keys (Play/Pause, Next Track, etc.) and handle them both using the old method AND using the new method so you'll get at least one of the events even if the user is running the Microsoft or Logitech keyboard software. You should only need to do this for those special keys.

**Comment by dudeinco** [comment]
Agreed ... Elpis media keys used to work, now they don't, so something has gone wrong on the recent upgrades - Surface Pro 3, no support for keys anymore without killing iType.  The solution until a solution is reached would be to grab the old code for the key handling and mix with the new updates.  ...Going to look to see what changed.

**Comment by seliver** [comment]
Ok, I'll have a look in it as well when get the time.

Alexey

Às 17:11 de 29/06/2015, dudeinco escreveu:

**Comment by dudeinco** [comment]
-Thanks - looks like the low level KB capture in KeyboardHook.cs (deprecated) was able to capture - 1.4.6 introduced Hotkeys.cs in place of the old class, which methods are unable to capture over Microsoft iType.  That's a starting point - I'd look more into it myself, but pulled a little thin ;)

**Comment by grojguy** [comment]
Any chance anyone has had a chance to look into this further?    Thank you!
